scarlet/psf.py

```diff
@@ -95,7 +144,8 @@ def fit_target_psf(psfs, func, init_values=None, extract_values=None):
     def reshape_func(*args):
         """Flatten the function output to work with `scipy.curve_fit`
         """
-        return func(*args).reshape(-1)
+        _args = [y, x] + list(args[1:])
```

Comment:
I'm not sure I follow. Can you explain this?

Reply:
This is to match the API of `scipy.curve_fit`, which requires the dependent variable as the first argument and the result to be a 1D array. The old version of the psf functions passed the meshgrid `(Y, X)` but for the trapezoid rule it is more convenient to use a single 1D array for `y` and another for `x` instead of a mesh grid, so I had to create a hack to make the `curve_fit` function work properly.